### PR TITLE
- Add test for sequencing.

### PR DESCRIPTION
**Pull request by baldersheim**
- Also sequence multiple operations to the same id as the originating request.

@bratseth PR

### CODE REVIEW

**Comment by baldersheim** [comment]
Test failure is unrelated.

File: docproc/src/main/java/com/yahoo/docproc/jdisc/messagebus/MbusRequestContext.java [hunk 1]
```diff
@@ -96,12 +97,16 @@ public void processingDone(List<Processing> processings) {
         }
         long inputSequenceId = requestMsg.getSequenceId();
         ResponseMerger responseHandler = new ResponseMerger(requestMsg, messages.size(), this);
+        int numMsgWithOriginalSequenceId = 0;
         for (Message message : messages) {
+            if (message.getSequenceId() == inputSequenceId) numMsgWithOriginalSequenceId++;
+        }
+        for (Message message : messages) {
+            String path = internalNoThrottledSourcePath;
+            if ((numMsgWithOriginalSequenceId == 1) && (message.getSequenceId() == inputSequenceId))
```
Comment (bratseth) [inline docproc/src/main/java/com/yahoo/docproc/jdisc/messagebus/MbusRequestContext.java:106]: Shouldn't this be numMsgWithOriginalSequenceId >= 1 ?

**Review by bratseth — COMMENTED** [review]
otherwise LGTM!

File: docproc/src/main/java/com/yahoo/docproc/jdisc/messagebus/MbusRequestContext.java (hunk 1 above)
Comment (baldersheim) [reply docproc/src/main/java/com/yahoo/docproc/jdisc/messagebus/MbusRequestContext.java:106]: No, you shall only continue on the original path if there is exactly one operation with the same id as the incoming operation. If there are multiple operations with the same id they must always be sequenced. If they are to the original id or not does not matter.

Comment (baldersheim) [reply docproc/src/main/java/com/yahoo/docproc/jdisc/messagebus/MbusRequestContext.java:106]: The if is actually the normal path :)

Comment (bratseth) [reply docproc/src/main/java/com/yahoo/docproc/jdisc/messagebus/MbusRequestContext.java:106]: 👍

**Pull request merged**